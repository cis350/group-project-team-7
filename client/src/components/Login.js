import React, { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";
import { TextField } from "@mui/material";
import toast from 'react-hot-toast';

// const serverHost = process.env.REACT_APP_SERVER_HOST;
// const serverPort = process.env.REACT_APP_SERVER_PORT;
const serverUrl = process.env.REACT_APP_SERVER_URL;

const Login = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const navigate = useNavigate();

  useEffect(() => {
    fetch(`${serverUrl}/get_current_user`, {
      method: "GET",
      credentials: 'include',
    }).then((res) => res.text())
      .then((resText) => {
        console.log(resText)
        if (resText) {
          console.log("wow")
        }
      });
  }, []);

  // Login function
  const handleLogin = () => {
    if (username.length === 0) {
      toast.error("Username must be filled")
    } else if (password.length === 0) {
      toast.error("Password must be filled")
    } else {
      fetch(`${serverUrl}/login?username=${username}&password=${password}`, {
        method: "POST",
        credentials: 'include'
      }).then((data) => {
        if (data.status === 201) {
          toast.success("Logged in!")
          navigate("/domestic");
        } else {
          toast.error("Username or password is incorrect")
        }
      }
      );
    }
  };

  return (
    <>
      <div style={{ height: "100vh", padding: "5vh" }} className="w-100">
        <div
          style={{ marginBottom: "12vh" }}
          className="d-flex justify-content-center"
        >
          {/* <img
            className="m-auto text-center text-light"
            src={"https://w7.pngwing.com/pngs/773/201/png-transparent-airplane-aircraft-flight-logo-airplane-blue-logo-flight-thumbnail.png"}
            width="25"
          ></img> */}
        </div>
        <div className="text-center mb-4 text-4xl font-semibold">Moder Patshala </div>
        <div className="text-center mb-4 text-m">Guiding Students and Parents to a Brighter Future Since 2004</div>
        <hr
            style={{
                color: "black",
                backgroundColor: "gray",
                height: 2,
                width: 550,
                margin: "auto",
                marginBottom: "20px",
            }}
        />
        <div className="mx-auto w-fit">
          <div>
            <TextField
              className="w-[24vw]"
              color="primary"
              id="outlined-required"
              variant="outlined"
              label="Username"
              defaultValue=""
              onChange={(e) => setUsername(e.target.value)}
            />
          </div>
          <div className="mt-2">
            <TextField
              className="w-[24vw]"
              color="primary"
              id="outlined-required"
              variant="outlined"
              label="Password"
              defaultValue=""
              onChange={(e) => setPassword(e.target.value)}
            />
          </div>
        </div>
        <div className="m-auto text-center mb-3">
          <button
            type="button"
            onClick={() => handleLogin()}
            className="text-white w-[24vw] bg-lime-900 mt-4 hover:bg-lime-950 hover:scale-105 active:scale-100 duration-150 font-medium rounded px-5 py-4 focus:outline-none">
            Continue
          </button>
        </div>
        <p className="m-auto text-center text-sm">
          {"Don't have an account? "}
          <a href="/signup" className="text-blue-500 text-decoration-none">
            Sign up
          </a>
        </p>
      </div>
    </>
  );
};

export default Login;
