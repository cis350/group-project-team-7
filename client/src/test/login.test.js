import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Login from './Login';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

// Mock server response for the login functionality
const server = setupServer(
  rest.post(`${process.env.REACT_APP_SERVER_URL}/login`, (req, res, ctx) => {
    return res(ctx.json({ status: 201 }));
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test('renders login inputs and a button', () => {
  render(<Router><Login /></Router>);
  
  // Check for input fields
  expect(screen.getByLabelText(/Username/i)).toBeInTheDocument();
  expect(screen.getByLabelText(/Password/i)).toBeInTheDocument();

  // Check for the login button
  expect(screen.getByRole('button', { name: /continue/i })).toBeInTheDocument();
});



// import { render, screen, fireEvent, waitFor } from '@testing-library/react';
// import { BrowserRouter } from 'react-router-dom';
// import Login from '../components/Login';
// import App from '../App.js'

// test('render login page', async () => {
//   render(
//       <BrowserRouter>
//         <Login />
//       </BrowserRouter>
//   );

//   const loginText = await screen.findByText(/login/i);
//   expect(loginText).toBeInTheDocument();


//   // Get sign-up link
//   // const signupLinkElement = screen.getByText(/Sign up/i);
  
//   // expect(signupLinkElement).toBeInTheDocument();
// })