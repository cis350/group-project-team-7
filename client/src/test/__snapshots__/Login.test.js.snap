// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-100"
    style="height: 100vh; padding: 5vh;"
  >
    <div
      class="d-flex justify-content-center"
      style="margin-bottom: 12vh;"
    />
    <div
      class="text-center mb-4 text-4xl font-semibold"
    >
      Moder Patshala 
    </div>
    <div
      class="text-center mb-4 text-m"
    >
      Guiding Students and Parents to a Brighter Future Since 2004
    </div>
    <hr
      style="color: black; background-color: gray; height: 2px; width: 550px; margin: auto auto 20px auto;"
    />
    <div
      class="mx-auto w-fit"
    >
      <div>
        <div
          class="MuiFormControl-root MuiTextField-root w-[24vw] css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="username-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-required"
            id="outlined-required-label"
          >
            Username
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-required"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="mt-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root w-[24vw] css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          data-testid="password-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-required"
            id="outlined-required-label"
          >
            Password
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-required"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      class="m-auto text-center mb-3"
    >
      <button
        class="text-white w-[24vw] bg-lime-900 mt-4 hover:bg-lime-950 hover:scale-105 active:scale-100 duration-150 font-medium rounded px-5 py-4 focus:outline-none"
        type="button"
      >
        Continue
      </button>
    </div>
    <p
      class="m-auto text-center text-sm"
    >
      Don't have an account? 
      <a
        class="text-blue-500 text-decoration-none"
        href="/signup"
      >
        Sign up
      </a>
    </p>
  </div>
</DocumentFragment>
`;
